docs/run_notebooks.py

```diff
@@ -0,0 +1,196 @@
+import io
+import os
+import json
+import time
+import pathlib
+import argparse
+import warnings
+import traceback
+from shutil import copyfile
+from contextlib import redirect_stderr, redirect_stdout
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+
+
+def execute_notebook(nb_test_dir, filename):
```

Comment:
would it be easier to run it via jupyter with `jupyter execute notebook.ipynb`?

https://docs.jupyter.org/en/latest/running.html#using-a-command-line-interface

Reply:
I considered that, but I figured it'd be easier to silence specific (expected) warnings and promote (and capture) others to exceptions when evaluating it this route.